test/widgets/content_test.dart

```diff
@@ -158,6 +166,78 @@ void main() {
     });
   });
 
+  group('Internal links', () {
+    Future<List<Route<dynamic>>> prepareContentWithNavigator(WidgetTester tester, {
+      required String html,
+    }) async {
+      addTearDown(testBinding.reset);
+      final pushedRoutes = <Route<dynamic>>[];
+      final testNavObserver = TestNavigatorObserver()
+        ..onPushed = (route, prevRoute) => pushedRoutes.add(route);
+      await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot(
+        streams: [eg.stream(streamId: 1, name: 'check')],
+      ));
+      final store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
+      final connection = store.connection as FakeApiConnection;
+      connection.prepare(json: newestResult(
+        foundOldest: true,
+        messages: [],
+      ).toJson());
+      await tester.pumpWidget(GlobalStoreWidget(child: MaterialApp(
+        navigatorObservers: [testNavObserver],
+        home: PerAccountStoreWidget(accountId: eg.selfAccount.id,
+          child: BlockContentList(nodes: parseContent(html).nodes)))));
+      await tester.pump();
+      assert(pushedRoutes.length == 1);
+      pushedRoutes.removeLast();
+      return pushedRoutes;
+    }
+
+    testWidgets('internal links are resolved: StreamNarrow', (tester) async {
+      final pushedRoutes = await prepareContentWithNavigator(tester,
+        html: '<p><a href="/#narrow/stream/1-check">stream</a></p>');
+
+      await tester.tap(find.text('stream'));
+      check(testBinding.takeLaunchUrlCalls()).isEmpty();
+      check(pushedRoutes).single.isA<WidgetRoute>()
+        .page.isA<MessageListPage>()
+        .narrow.equals(const StreamNarrow(1));
+    });
+
+    testWidgets('internal links are resolved: TopicNarrow', (tester) async {
+      final pushedRoutes = await prepareContentWithNavigator(tester,
+        html: '<p><a href="/#narrow/stream/1-check/topic/my.20topic">topic</a></p>');
```

Comment:
Having several versions like this for different types of narrow is OK, but FYI I'd say they're of low value — we already have extensive tests of the parsing that leads to the different types of narrow, over in `test/model/internal_link_test.dart` (from #305). The key things to test here are (a) some internal link, and (b) some link that resembles an internal link but is invalid, because those are the cases that this code-under-test knows the difference between.